webpack/scenes/Hosts/ChangeContentSource/components/Hosts.js

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  GridItem,
+  Label,
+  Modal,
+  ModalVariant,
+  List, ListItem,
+  SearchInput,
+} from '@patternfly/react-core';
+
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanUrl } from 'foremanReact/common/helpers';
+
+import PropTypes from 'prop-types';
+
+const Hosts = ({
+  contentHosts, hostsWithoutContent,
+}) => {
+  const [modal, handleModal] = useState(false);
+  const [modalHosts, setModalHosts] = useState([]);
+  const [modalTitle, setModalTitle] = useState('Hosts');
+  const [search, setSearch] = useState('');
+
+  const loadModal = (items, title) => {
+    setModalHosts(items);
+    setModalTitle(title);
+    setSearch('');
+    handleModal(true);
+  };
+
+  return (
+    <>
+      <GridItem span={7}>
+        {hostsWithoutContent.length > 0 &&
+        <p>
+          { __('Some hosts are not registered as Content Hosts and will be ignored.') }
+        </p>
+}
+        {contentHosts.length > 0 && (<Label color="green" href="#" onClick={() => loadModal(contentHosts.map(h => h.name), __('Hosts to update'))}>{__('Hosts to update')}: {contentHosts.length}</Label>)}
+        {' '}
+        {hostsWithoutContent.length > 0 && (<Label color="orange" href="#" onClick={() => loadModal(hostsWithoutContent, __('Ignored hosts'))}>{__('Ignored hosts')}: {hostsWithoutContent.length}</Label>)}
+
+      </GridItem>
+      <Modal
+        variant={ModalVariant.small}
+        title={modalTitle}
+        position="top"
+        isOpen={modal}
+        onClose={() => handleModal(false)}
+      >
+        <SearchInput
+          placeholder="Find by name"
+          value={search}
+          onChange={v => setSearch(v)}
+          onClear={() => setSearch('')}
+        />
+        <List isPlain isBordered className="margin-top-16">
+          {(search ? modalHosts.filter(h => (`${h}`).includes(search)) : modalHosts).map(h => (
+            <ListItem key={h}>
+              <Button
+                component="a"
+                href={foremanUrl(`/hosts/${h}`)}
```

Comment:
This links to the legacy host detail page, even though my "New host details UI" setting is turned on.